components/transaction-chart/index.js

```diff
@@ -40,13 +40,12 @@ function getDataset(transactionData) {
 
 const TransactionChart = (props) => {
   const [chartData, setChartData] = useState({});
-  const chart = () => {
-    setChartData(getDataset(props.transactionChartData));
-  };
-
   useEffect(() => {
+    const chart = () => {
+      setChartData(getDataset(props.transactionChartData));
+    };
     chart();
-  }, [chart]);
+  }, [props.transactionChartData]);
```

Comment:
NIT: Best practice will be to use destructuring

Reply:
You are right, I will update that